src/components/MAPI/apps/AppList/utils.tsx

```diff
@@ -0,0 +1,327 @@
+ * @param selectedAppCatalogs - The catalogs that are currently enabled in the sidebar.
+ * @param appCatalogErrors - Errors that occured during loading each catalog.
+ */
+export function mapAppCatalogsToFacets(
+  appCatalogs: applicationv1alpha1.IAppCatalog[] = [],
+  selectedAppCatalogs: Record<string, boolean> = {},
+  appCatalogErrors: Record<string, string> = {}
+): IFacetOption[] {
+  return appCatalogs.sort(compareAppCatalogs).map((appCatalog) => {
+    const uiTitle = computeAppCatalogUITitle(appCatalog);
+
+    return {
+      value: appCatalog.metadata.name,
+      checked: selectedAppCatalogs.hasOwnProperty(appCatalog.metadata.name),
+      label: (
+        <CatalogLabel
+          catalogName={uiTitle}
+          iconUrl={appCatalog.spec.logoURL}
+          description={appCatalog.spec.description}
+          error={appCatalogErrors[appCatalog.metadata.name]}
+        />
+      ),
+    };
+  });
+}
+
+/**
+ * Search through the apps and find one that is in the
+ * selected catalogs, and that matches the search query.
+ * @param searchQuery
+ * @param indexApps
+ */
+export function filterAppCatalogIndexApps(
+  searchQuery: string,
+  indexApps: IAppCatalogIndexApp[],
+  selectedAppCatalogs: Record<string, boolean>
+) {
+  const normalizedQuery = searchQuery.trim().toLowerCase();
+
+  return indexApps.filter((indexApp) => {
+    if (!selectedAppCatalogs.hasOwnProperty(indexApp.catalogName)) return false;
+
+    if (normalizedQuery.length < 1) return true;
+
+    if (indexApp.versions.length < 1) return false;
+    const version = indexApp.versions[0];
+
+    switch (true) {
+      case version.name.toLowerCase().includes(normalizedQuery):
+      case version.description &&
+        version.description.toLowerCase().includes(normalizedQuery):
+      case version.keywords &&
+        version.keywords.join(',').toLowerCase().includes(normalizedQuery):
+        return true;
+
+      default:
+        return false;
+    }
+  });
+}
+
+function makeAppPath(
+  name: string,
+  version: string,
+  catalogName: string
+): string {
+  return RoutePath.createUsablePath(AppsRoutes.AppDetail, {
+    app: name,
+    catalogName,
+    version,
+  });
+}
+
+function compareVersions(
+  a: IAppCatalogIndexAppVersion,
+  b: IAppCatalogIndexAppVersion
+) {
+  return compare(b.version, a.version);
+}
+
+interface IAppCatalogIndexResponse {
+  apiVersion: string;
+  entries: Record<string, IAppCatalogIndexAppVersion[]>;
+}
+
+/**
+ * Fetch the catalog index from the URL saved in the catalog CR.
+ * @param fetchFunc
+ * @param _auth
+ * @param catalog
+ */
+async function fetchAppCatalogIndex(
+  fetchFunc: (input: RequestInfo, init?: RequestInit) => Promise<Response>,
+  _auth: IOAuth2Provider,
+  catalog: applicationv1alpha1.IAppCatalog
+): Promise<IAppCatalogIndex> {
+  const url = normalizeAppCatalogIndexURL(catalog.spec.storage.URL);
+
+  // Enforce client-side CORS.
+  const response = await fetchFunc(url, { mode: 'cors' });
```

Comment:
We need to fetch the catalog index from GitHub, as apps include more info than via the K8s API (e.g. images)